app/controllers/registrations_controller.rb

```
@@ -27,4 +26,17 @@ def setup_user
     @user = user_from_params
     @user.confirmation_token = Clearance::Token.new
   end
+
+  def user_params # rubocop:disable Metrics/MethodLength
+    if params[:user]
+      params[:user].permit(
+        :email,
+        :first_name,
+        :last_name,
+        :password
+      )
+    else
+      {}
```

if params[:user] is not found, we don't want it to fail silently.  Are you handling when the caller receives a {} ?